src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/AbstractMultiLifetimeComponent.cs

```diff
@@ -47,38 +59,37 @@ public async Task LoadAsync()
             await LoadCoreAsync();
         }
 
-        private async Task LoadCoreAsync()
+        public async Task LoadCoreAsync()
         {
-            TaskCompletionSource<object> loadedSource = null;
-            IMultiLifetimeInstance instance;
+            JoinableTask initializeAsyncTask;
+
             lock (_lock)
```

Comment:
Could we use something like AsyncReaderWriterLock? I agree that this logic looks correct (as much as I can understand it).   But do we really want to be blocking threadpool threads on initialization?

Reply:
As mentioned on the PR that does this - this is lock has basically zero contention.